### PR TITLE
new challenge: reverse_words

### PR DESCRIPTION
**Pull request by mag6367**
Hi again Donne, 

Let me know if the challenge looks better! 

I hope it does :) 

I will also switch gears to more complex challenges once I get good at making these notebooks 

-Marco

### CODE REVIEW

**Comment by mag6367** [comment]
Hi Donne, 

So sorry to bother you again but I was working on some commits and I accidentally pushed to the master. I wasn't intending on pull requesting with more than one challenge. 

-Marco

**Comment by donnemartin** [comment]
Hi Marco,

Thanks for the heads up.  I'll try to review this over the weekend.

-Donne

**Comment by donnemartin** [comment]
Hi Marco,

I'll place this in the "staging" directory for some tweaks and polishing.  Please review the comments I made here:

https://github.com/donnemartin/interactive-coding-challenges/pull/19

Thanks for the contribution!

-Donne

**Comment by mag6367** [comment]
Hi Donne, 

Will do!

Thank you :)

-Marco

**Comment by mag6367** [comment]
Hi Donne, 

How are you?

For complex solutions like the binary tree implementation, would you like comments? I noticed that it can be hard to follow because it is a lot of code. I'm working on a Graph implementation and hoping to improve the binary tree so it can be released on the main folder. 

Thanks 

-Marco

**Comment by donnemartin** [comment]
Hi Marco,

Great to hear from you again :)

I think comments help for some more complex chunks of code.  I'd be careful about adding too many comments though--I feel generally Python is pretty readable, almost pseudocode.

-Donne
